src/averageAges.js

```diff
@@ -55,7 +67,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person =>
+    !onlyWithSon
+      ? people.find(child => child.name === person.mother)
+      : person.sex === 'm' && people.find(child =>
+        child.name === person.mother));
+
+  const mothers = people.filter(person =>
+    person.sex === 'f'
+    && (onlyWithSon === undefined
```

Comment:
Instead of checking for 'onlyWithSon === undefined', use '!onlyWithSon' for better readability.